__main__.py

```diff
@@ -12,28 +12,40 @@
 from datetime import datetime, timedelta
 from functools import partial
 
+import iso8601
 from mws import mws
 from pymongo import MongoClient
 
 from .orders import set_order_items, generate_orders
 from .utils import make_ratelimit_aware
 
-MARKETPLACEIDS = os.environ["MARKETPLACEIDS"].split(",")
-DAYS_AGO = 1 if "DAYS_AGO" not in os.environ else int(os.environ["DAYS_AGO"])
-
 orders_api = mws.Orders(
     os.environ["AWS_ACCESS_KEY"],
     os.environ["MWS_SECRET_KEY"],
     os.environ["MWS_SELLERID"],
     region=os.environ["REGION"],
 )
 
-start_date = datetime.now() - timedelta(days=DAYS_AGO)
+MARKETPLACEIDS = os.environ["MARKETPLACEIDS"].split(",")
+
+START_DATE = None
+END_DATE = None
+if "START_DATE" in os.environ:
+    START_DATE = iso8601.parse_date(os.environ["START_DATE"]).isoformat()
+    if "END_DATE" in os.environ:
+        END_DATE = iso8601.parse_date(os.environ["END_DATE"]).isoformat()
+else:
+    DAYS_AGO = 1 if "DAYS_AGO" not in os.environ else int(os.environ["DAYS_AGO"])
```

Comment:
Gah. That's ugly. Does python have some kind of ternary or short-circuiting to leverage here?

Reply:
You're _looking_ at Python's ternary, which is short-circuiting. In case it isn't clear, Guido doesn't want you to use it.

```
In [1]: 1 if True else BOOM()
Out[1]: 1
```